File: spring-rest-configuration-autodiscover.java/src/main/java/demo/service/example/ExamplePropertyStringService.java
package demo.service.example;

import demo.service.core.Code;
import demo.service.core.PropertyStringService;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class ExamplePropertyStringService extends PropertyStringService {

    @Value("${example.property.string}")
    private String defaultValue;

    @Override
    public Code getCode() {
        return Code.EXAMPLE_STRING_PROPERTY;
    }

    @Override
    public String getDefaultValue() {
        return defaultValue;
    }
}

File: spring-rest-configuration-autodiscover.groovy/src/main/kotlin/demo/service/example/ExamplePropertyStringService.kt
package demo.service.example

import demo.service.core.Code
import demo.service.core.PropertyStringService
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyStringService : PropertyStringService() {

    @Value("\${example.property.string}")
    private val defaultValue: String? = null

    override fun getCode() = Code.EXAMPLE_STRING_PROPERTY

    public override fun getDefaultValue() = defaultValue!!
}
